crates/rolldown/src/dev/watcher_event_service.rs

```diff
@@ -38,20 +39,32 @@ impl WatcherEventService {
       match msg {
         WatcherEventServiceMsg::FileChange(file_change_result) => match file_change_result {
           Ok(batched_events) => {
-            let changed_files = batched_events
-              .into_iter()
-              .flat_map(|batched_event| match &batched_event.detail.kind {
-                notify::EventKind::Modify(_modify_kind) => batched_event.detail.paths,
-                _ => {
-                  vec![]
-                }
-              })
-              .collect::<Vec<_>>();
-
-            self.build_driver.register_changed_files(changed_files).await;
+            // TODO: using a IndexSet here will cause changes like [a.js, b.js, a.js] to be [a.js, b.js].
+            // Not sure if we want this behavior for hmr scenario.
+            let mut changed_files = FxIndexSet::default();
```

Comment:
Does the order of `changed_files` matter?

Reply:
For generating hmr updates, yes. Because updates are generated per file, sure we don't want to generate updates for later changed file and then generate for the first one.

But currently I can't think of clear situation how it would hit this scenario.

Reply:
If the HMR takes time (maybe due to some plugins?), the changes will be queued. In that case, I guess it would happen.